src/native/http_connection_manager.c

```diff
@@ -353,21 +353,15 @@ static void s_on_http_conn_acquisition_callback(
         error_code,
         aws_error_str(error_code));
 
-    jobject acquire_connection_future = (*env)->NewLocalRef(env, binding->java_acquire_connection_future);
-    if (acquire_connection_future != NULL) {
-        (*env)->CallStaticVoidMethod(
-            env,
-            http_client_connection_properties.http_client_connection_class,
-            http_client_connection_properties.on_connection_acquired_method_id,
-            acquire_connection_future,
-            (jlong)binding,
-            jni_error_code);
+    (*env)->CallStaticVoidMethod(
+        env,
+        http_client_connection_properties.http_client_connection_class,
+        http_client_connection_properties.on_connection_acquired_method_id,
+        binding->java_acquire_connection_future,
+        (jlong)binding,
+        jni_error_code);
 
-        AWS_FATAL_ASSERT(!aws_jni_check_and_clear_exception(env));
-        (*env)->DeleteLocalRef(env, acquire_connection_future);
```

Comment:
trivial: worth deleting `binding->java_acquire_connection_future` after firing the callback? we don't need it anymore, right?

Reply:
Good point.